app/src/main/java/com/cornellappdev/transit/ui/components/LocationItem.kt

```diff
@@ -0,0 +1,138 @@
+ * @param label The label for the item
+ * @param sublabel The sublabel for each item
+ * @param editing Whether or nto the list if currently being edited
+ * @param onClick The Function to run when the add buttom is clicked
+ */
+@Composable
+fun LocationItem(
+    image: Painter,
+    editImage: Painter,
+    label: String,
+    sublabel: String,
+    editing: Boolean,
+    onClick: () -> Unit
+) {
+    Column(
+        modifier = Modifier
+            .width(88.dp)
+            .height(86.dp)
+            .padding(horizontal = 8.dp)
+    ) {
+        if (editing && label != "Add") {
+            Box(
+                modifier = Modifier
+                    .size(48.dp)
+                    .align(Alignment.CenterHorizontally)
+            ) {
+                Image(
+                    editImage,
+                    contentDescription = "Place",
+                    modifier = Modifier
+                        .align(Alignment.Center)
+                        .size(48.dp)
+
+                )
+                Image(
+                    painter = painterResource(id = R.drawable.remove_icon),
+                    contentDescription = "AdRemoved",
+                    modifier = Modifier
+                        .align(Alignment.TopEnd)
+                        .size(20.dp)
+
+                )
+
+            }
+        } else {
+            Box(
+                modifier = Modifier
+                    .height(48.dp)
+                    .width(48.dp)
+                    .align(Alignment.CenterHorizontally)
+            ) {
+                Image(
+                    image,
+                    contentDescription = "Place",
+                    modifier = Modifier
+                        .size(48.dp)
+                        .align(Alignment.Center)
+                        .clickable(onClick = onClick)
+                        .takeIf { label == "Add" } ?: Modifier
+                )
+
+                if (label == "Add") {
+                    Image(
+                        editImage,
+                        contentDescription = "Place",
+                        modifier = Modifier
+                            .align(Alignment.Center)
+                            .size(30.dp)
+                    )
+                }
+            }
+        }
+        Text(
+            label,
+            fontSize = 12.sp,
+            color = if (label == "Add") TransitBlue else Color.Black,
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .padding(top = 8.dp, bottom = 4.dp)
+                .height(14.dp)
+                .width(72.dp)
+        )
+
+        Text(
+            sublabel,
+            color = Color.Gray,
+            fontSize = 10.sp,
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .align(Alignment.CenterHorizontally)
+                .height(12.dp)
+                .width(72.dp)
```

Comment:
I see here that you're fixing the `height` of each of these `Text`s. If you look at the Figma, it suggests that although the text does indeed fit this height, it's automatically computed rather than fixed:

![image](https://github.com/cuappdev/ithaca-transit-compose/assets/47724806/7c6158ab-c77e-4768-aef0-bd9b661b5639)

With cases like these, we usually try not to hardcode component size unless there's some good reason to do so. Otherwise, while we will certainly pay attention to details like font size, font weight, etc, we typically ignore explicit widths and heights and instead just structure the component such that the height would automatically match (or close enough to it).